File: pset6/readability/readability.py
from cs50 import get_string
import re

def main():
    sentence = get_string("Text: ")
    grade(sentence)
    
def grade(sentence):
    count_letters = 0
    
    for i in range(len(sentence)):
        if sentence[i].isalpha():
            count_letters += 1
    
    count_words = len(sentence.split())
    count_sentences = len(re.split("[.!?]", sentence)) - 1
    
    L = round((count_letters * 100) / count_words, 2)
    S = round((count_sentences * 100) / count_words, 2)
    
    
    index = 0.0588 * L - 0.296 * S - 15.8
    index = round(index)
    
    if (index >= 16):
        print("Grade 16+")
    elif index <= 1:
        print("Before Grade 1")
    else:
        print("Grade " + str(index))

    
if __name__ == "__main__":
    main()

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
#include <string.h>

int count_letters(string x) ;
int count_words(string x) ;
int count_sentence(string x);


int main(void)
{
    string text = get_string("Text: ");
    int letter = count_letters(text);
    int word = count_words(text);
    int sentence = count_sentence(text);
    
    float L = (letter / (float)word)*100;
    float S = (sentence / (float)word)*100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    
    
    if(index >= 1 && index <16)
    {
        printf("Grade %i\n", index);
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    
}

//My Functions

int count_letters(string x)
{
    int sum = 0 ;
    int n = strlen(x);
    for (int i = 0 ; i < n ; i++)
    {
        if ((x[i] >= 'a' && x[i] <= 'z') || (x[i] >= 'A' && x[i] <= 'Z'))
        {
            sum = sum + 1 ;
        }
    }
    return sum ;
}

int count_words(string x)
{
    int sum = 1 ;
    int n = strlen(x);
    for (int i = 0 ; i < n ; i++)
    {
        if (x[i] == ' ')
        {
            sum = sum +1 ;
        }
    }
    return sum ;
}

int count_sentence(string x)
{
    int sum = 0 ;
    int n = strlen(x);
    
    for (int i = 0 ; i < n ; i++)
    {
        if (x[i] == '.' || x[i] == '!' || x[i] == '?')
        {
            sum = sum + 1 ;
        }
    }
    return sum ;
}